public/js/graph/react_graph.js

```diff
@@ -261,28 +261,73 @@ var Graph = React.createClass({
     },
 
     nodeMouseEnter: function (event) {
-        var courseID = event.currentTarget.id;
-        var currentNode = this.refs.nodes.refs[courseID];
+        var courseId = event.currentTarget.id;
+        var currentNode = this.refs.nodes.refs[courseId];
         currentNode.focusPrereqs(this);
-
-        // Old hover modal code
-        if ($('.modal').length === 0) {
-            $('.tooltip-group').remove();
-            tooltip.displayTooltip(courseID);
-        }
+				
+        clearAllTimeouts();
+			
+				var infoBox = this.refs.infobox;
+				//TODO: adjust xPos and yPos
+				var xPos = currentNode.props.JSON.pos[0];
+				var yPos = currentNode.props.JSON.pos[1];
+				var rightSide = xPos > 222;
+				// The tooltip is offset with a 'padding' of 5.
+				if (rightSide) {
+						xPos = parseFloat(xPos) - 65;
+				} else {
+						xPos = parseFloat(xPos) +
+								parseFloat(currentNode.props.JSON.width) + 5;
+				}
+
+				yPos = parseFloat(yPos);
+				
+				infoBox.setState({xPos: xPos,
+													yPos: yPos,
+													nodeId: courseId,
+												 	showInfobox: true});
     },
 
     nodeMouseLeave: function (event) {
-        var courseID = event.currentTarget.id;
-        var currentNode = this.refs.nodes.refs[courseID];
+        var courseId = event.currentTarget.id;
+        var currentNode = this.refs.nodes.refs[courseId];
         currentNode.unfocusPrereqs(this);
-
-        // Old hover modal code
-        if ($('.modal').length === 0) {
-            tooltip.removeTooltip(this);
-        }
+				
+				var infoBox = this.refs.infobox;
+				
+				var timeout = setTimeout(function () {
+						infoBox.setState({showInfobox: false});
+				}, 400);
+
+				timeouts.push(timeout);
```

Comment:
You probably know this, but `timeouts` should be state of some component. (And one more global variable would be removed!)